### PR TITLE
redisDB

### PR DESCRIPTION
**Pull request by shay123g**

### CODE REVIEW

**Comment by shay123g** [comment]
close #29

File: src/errors/invalidUrl.ts [hunk 1]
```diff
@@ -1,3 +1,9 @@
 import createError from 'fastify-error'
 
 export const InvalidUrl = createError('ERR_INVALID_URL', `Invalid Url.`, 400)
+export const InvalidPayloadError = createError('ERR_INVALID_URL_PAYLOAD', `payload include invalid fields`, 400)
+export const IncompatibleResponse = createError(
+	'INCOMPATIBLE_RESPONSE',
+	'unexpected object type received from server',
+	undefined,
```
Comment (Danielg212) [inline src/errors/invalidUrl.ts:8]: I think this should be 400 code

File: src/helpers/redisHelper.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+		this.urlWithInfo.urlId = id
+		if (!options.withInfo) {
+			return this.url
+	}
+		else {
+			return this.urlWithInfo
+		}
+	}
+
+	generateUrl(key: string, value: string): void{
```
Comment (Danielg212) [inline src/helpers/redisHelper.ts:44]: can you abstract this method? - remove the switch case and use more abstract method

File: src/services/storage/drivers/redis/index.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+		}
+
+		async save(url: UrlWithInformation): Promise<StoredUrl> {
+			if (url.id === 'undefined' || url.url === 'undefined') throw new InvalidUrl()
+			const storedUrlWithInfo = {
+				id: cryptoRandomString({ length: 6, type: 'url-safe' }),
+				url: url.url,
+				createdAt: new Date().toISOString(),
+				updatedAt: new Date().toISOString(),
+				ip: url.ip != '' ? url.ip : 'undefined',
```
Comment (Danielg212) [inline src/services/storage/drivers/redis/index.ts:60]: you can write this as `url.ip?url.ip: undefined `

**Review by Danielg212 — COMMENTED** [review]
try to reduce the switch case to more abstract method

File: src/services/storage/drivers/redis/types.ts [hunk 4]
```diff
@@ -0,0 +1,2 @@
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface RedisStorageDriverConfig {}
```
Comment (SnirShechter) [inline src/services/storage/drivers/redis/types.ts:2]: The Redis driver doesn't need any configuration? what about the URL? possible credentials?

**Review by SnirShechter — CHANGES_REQUESTED** [review]

**Pull request closed without merging**